taichi/ir/expr.h

```diff
@@ -134,4 +128,16 @@ Expr bit_cast(const Expr &input) {
   return taichi::lang::bit_cast(input, get_data_type<T>());
 }
 
+// like Expr::Expr, but allows to explicitly specify the type
+template <typename T>
+Expr value(const T &val) {
+  return Expr(val);
+}
+
+Expr expr_rand(DataType dt);
```

Comment:
Why leave `expr_rand` and `value` here, but put other expressions like `snode_append` in `frontend.h`?

Reply:
done - now `frontend.h` is purged.